controllers/consoleplugin/consoleplugin_objects.go

```diff
@@ -112,6 +113,55 @@ func buildArgs(desired *flowsv1alpha1.FlowCollectorConsolePlugin, desiredLoki *f
 }
 
 func (b *builder) podTemplate(cmDigest string) *corev1.PodTemplateSpec {
+	volumes := []corev1.Volume{{
+		Name: secretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	}, {
+		Name: configVolume,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: configMapName,
+				},
+			},
+		},
+	},
+	}
+
+	volumeMounts := []corev1.VolumeMount{{
+		Name:      secretName,
+		MountPath: "/var/serving-cert",
+		ReadOnly:  true,
+	},
+		{
+			Name:      configVolume,
+			MountPath: configPath,
+			ReadOnly:  true,
+		}}
+
+	args := []string{
+		"-cert", "/var/serving-cert/tls.crt",
+		"-key", "/var/serving-cert/tls.key",
+		"-loki", querierURL(b.desiredLoki),
+		"-loki-labels", strings.Join(constants.LokiIndexFields, ","),
+		"-loki-tenant-id", b.desiredLoki.TenantID,
+		"-loki-skip-tls", "true",
+		"-loglevel", b.desired.LogLevel,
+		"-frontend-config", configPath + configFile}
+
+	if b.desiredLoki != nil && b.desiredLoki.TLS.Enable {
```

Comment:
Maybe I'm wrong but I think the `InsecureSkipVerify` and the CA cert shouldn't be mutually exclusive. The loki server might expect TLS encryption (thus needs CA cert), but still, `InsecureSkipVerify` can be used (in this case it is the client that skips certificate verification, but that doesn't mean it wouldn't encrypt communications).
At least that's my understanding. So if that's correct, there shouldn't be an `else` clause here for CA.

Reply:
InsecureSkipVerify does not mean not using TLS it means no checking that the certificate used by the remote server is properly signed.
The CA cert is used for the same reason, it is used to validate the remote server certificate.

Reply:
> InsecureSkipVerify does not mean not using TLS it means no checking that the certificate used by the remote server is properly signed.

That's what I say :)

>  The CA cert is used for the same reason, it is used to validate the remote server certificate. 

But the CA cert contains the server public key. So if TLS is enabled then I think you _need_ this cert for encryption, even if InsecureSkipVerify is true

Reply:
(If this is wrong then I need to fix that on Kafka side, because that's how I did)

Reply:
Ok, I've read a little more about TLS and I think you're right :)